vivisect/analysis/generic/codeblocks.py

```diff
@@ -75,8 +75,11 @@ def analyzeFunction(vw, funcva):
                 brefs.append((va, False))
                 break
 
-            op = vw.parseOpcode(va)     # parseOpcode() pulls arch from the location db, if exists
-            mnem[op.mnem] += 1
+            try:
+                op = vw.parseOpcode(va)     # parseOpcode() pulls arch from the location db, if exists
+                mnem[op.mnem] += 1
+            except Exception:
+                pass
```

Comment:
i agree with the desire and your goals...  completing the emulation pass is important for many analysis things.  @rakuy0 and i have discussed this a bit in recent months.  

the problem is, how big is the opcode that failed to parse?  how do we calculate nextva?

my next question:  if its got a valid lsize, how did that location of type LOC_OP get created if it didn't parse the opcode?

a different option:  perhaps a "break" to get out of the current loop and start another iteration through the outer loop (instead of bailing on all analysis).

better yet:  can you help us fix the opcode that didn't parse correctly?  that PR would be the best help.  it doesn't seem like it's as big an impact, but it is a much more valuable impact over time.  the quality of our decoders and emulators is the highest importance.

Reply:
also, keep in mind that "pass" on exception without any sort of logging or emulog'ing loses potentially valuable information that something broken and what.  i've done this a lot, so i cringe saying it.

thanks!

Reply:
Logging is a good idea - I added it.

I will try to put up a separate PR with opcode corrections for some cases that I see.

Even if there was perfect opcode disassembly, code flow analysis sometimes explores where there is no legit code and an InvalidInstruction exception happens.  For instance, sample 0396166eafeaa4ea136dc20999e89974 is explored at 0x4018a7.  It may be that the codeblocks analysis does nothing in this case, but I still like catching the exception (and logging it).

Reply:
i fully agree about the value of catching and logging the exception.
the problem here is still that we don't know how big the opcode might have been so we may continue emulating completely bad code from this point.  "break"ing from here should terminate this emulation loop, and draw the next one from the queue.  instead of `pass`, put `break`, and i think we'll be gtg.

Reply:
```suggestion
                break
```

Reply:
replaced "pass" with "break"

By the way, upon closer inspection of my regression results, this is only happening on ELF files, not PE or Mach-O.